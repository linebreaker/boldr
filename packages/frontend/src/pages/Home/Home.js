/* @flow */

import React from 'react';
import Helmet from 'react-helmet';
import styled from 'styled-components';
// internal
import Headline from '@boldr/ui/Headline';
import Paragraph from '@boldr/ui/Paragraph';
import { Grid, Row, Col } from '@boldr/ui/Layout';

const HomeHero = styled.div`
  background-color: #243140;
  height: 450px;
  width: 100%;
`;

class Home extends React.Component {
  render() {
    return (
      <div>
        <Helmet title="Home" />
        <HomeHero>
          <Headline type="h1">Meet Boldr.</Headline>
        </HomeHero>
        <Grid>
          <Row>
            <Col xs={12} md={4}>
              <Headline type="h2">Placeholder 1</Headline>
              <Paragraph>
                This proposal introduces an Observable type to the ECMAScript standard library. The
                Observable type can be used to model push-based data sources such as DOM events,
                timer intervals, and sockets.
              </Paragraph>
            </Col>
            <Col xs={12} md={4}>
              <Headline type="h2">Placeholder 2</Headline>
              <Paragraph>
                The iterator interface (introduced in ECMAScript 2015) is a sequential data access
                protocol which enables the development of generic and composable data consumers and
                transformers. Their primary interface is a next() method which returns a value, done
                tuple, where done is a boolean indicating whether the end of the iterator has been
                reached, and value is the yielded value in the sequence.
              </Paragraph>
            </Col>
            <Col xs={12} md={4}>
              <Headline type="h2">Placeholder 3</Headline>
              <Paragraph>
                In ECMAScript, a realm consists of a global object and an associated set of
                primordial objects -- mutable objects like Array.prototype that must exist before
                any code runs. Objects within a realm implicitly share these primordials and can
                therefore easily disrupt each other by primordial poisoning -- modifying these
                objects to behave badly. This disruption may happen accidentally or maliciously.
                Today, in the browser, realms can be created via same origin iframes.
              </Paragraph>
            </Col>
          </Row>
        </Grid>
      </div>
    );
  }
}

export default Home;
