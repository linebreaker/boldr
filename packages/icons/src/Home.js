import React from 'react';
import Icon from './Icon';

/*eslint-disable */
const Home = (props) => (
  <Icon viewBox="0 0 24 24" {...props}>
    <g>
      <path d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z" />
      <path d="M9 22V12h6v10" />
    </g>
  </Icon>
);
/*eslint-enable */

export default Home;
