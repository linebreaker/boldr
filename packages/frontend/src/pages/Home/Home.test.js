import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from './Home';

describe('<Home />', () => {
  const wrapper = shallow(<Home />);
  it('renders <Home /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
