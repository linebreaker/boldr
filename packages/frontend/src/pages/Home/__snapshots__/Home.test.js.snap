// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders snapshot 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
    title="Home"
  />
  <styled.div>
    <Headline
      type="h1"
    >
      Meet Boldr.
    </Headline>
  </styled.div>
  <Grid
    componentClass="div"
    fluid={false}
  >
    <Row
      componentClass="div"
    >
      <Col
        componentClass="div"
        md={4}
        xs={12}
      >
        <Headline
          type="h2"
        >
          Placeholder 1
        </Headline>
        <Paragraph
          isLead={false}
          isLight={false}
        >
          This proposal introduces an Observable type to the ECMAScript standard library. The Observable type can be used to model push-based data sources such as DOM events, timer intervals, and sockets.
        </Paragraph>
      </Col>
      <Col
        componentClass="div"
        md={4}
        xs={12}
      >
        <Headline
          type="h2"
        >
          Placeholder 2
        </Headline>
        <Paragraph
          isLead={false}
          isLight={false}
        >
          The iterator interface (introduced in ECMAScript 2015) is a sequential data access protocol which enables the development of generic and composable data consumers and transformers. Their primary interface is a next() method which returns a value, done tuple, where done is a boolean indicating whether the end of the iterator has been reached, and value is the yielded value in the sequence.
        </Paragraph>
      </Col>
      <Col
        componentClass="div"
        md={4}
        xs={12}
      >
        <Headline
          type="h2"
        >
          Placeholder 3
        </Headline>
        <Paragraph
          isLead={false}
          isLight={false}
        >
          In ECMAScript, a realm consists of a global object and an associated set of primordial objects -- mutable objects like Array.prototype that must exist before any code runs. Objects within a realm implicitly share these primordials and can therefore easily disrupt each other by primordial poisoning -- modifying these objects to behave badly. This disruption may happen accidentally or maliciously. Today, in the browser, realms can be created via same origin iframes.
        </Paragraph>
      </Col>
    </Row>
  </Grid>
</div>
`;
